### PR TITLE
fix(components): update imageURL

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/KFLUXBUGS-1379

## Description

Updates component image to show IMAGE_URL from latest pipeline and fall back to containerImage if unavailable.

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2024-07-31 at 8 50 58 PM

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/Components/ComponentDetails/tabs/ComponentDetails.tsx [hunk 1]
```diff
@@ -8,17 +8,31 @@ import {
...
+const RESULT_NAME = 'IMAGE_URL';
+
 const ComponentDetails: React.FC<React.PropsWithChildren<ComponentDetailsProps>> = ({
   component,
 }) => {
+  const { namespace } = useWorkspaceInfo();
+  const [pipelineRun, pipelineRunLoaded, error] = useLatestSuccessfulBuildPipelineRunForComponent(
+    namespace,
+    component.metadata.name,
+  );
```
Comment (sahil143) [inline src/components/Components/ComponentDetails/tabs/ComponentDetails.tsx:31]: We need to use the latest push build here. If a PR build is the latest, this hook will return that

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: src/components/Components/ComponentDetails/tabs/ComponentDetails.tsx (hunk 1 above)
Comment (abhinandan13jan) [reply src/components/Components/ComponentDetails/tabs/ComponentDetails.tsx]: updated

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: src/hooks/usePipelineRuns.ts [hunk 2]
```diff
@@ -193,6 +193,30 @@ export const useLatestBuildPipelineRunForComponent = (
...
+            [PipelineRunLabel.COMMIT_EVENT_TYPE_LABEL]: PipelineRunEventType.PUSH,
+          },
+        },
+        limit: 1,
+      }),
+      [componentName],
+    ),
+  ) as unknown as [PipelineRunKind[], boolean, unknown];
+
+  return React.useMemo(() => [result[0]?.[0], result[1], result[2]], [result]);
```
[the remark this answers is not in the document]
Comment (abhinandan13jan) [reply src/hooks/usePipelineRuns.ts]: updated

File: src/hooks/usePipelineRuns.ts [hunk 3]
```diff
@@ -193,6 +193,30 @@ export const useLatestBuildPipelineRunForComponent = (
...
+            [PipelineRunLabel.PIPELINE_TYPE]: PipelineRunType.BUILD,
+            [PipelineRunLabel.COMPONENT]: componentName,
+            [PipelineRunLabel.COMMIT_EVENT_TYPE_LABEL]: PipelineRunEventType.PUSH,
+          },
+        },
+        limit: 1,
+      }),
+      [componentName],
+    ),
+  ) as unknown as [PipelineRunKind[], boolean, unknown];
```
[the remark this answers is not in the document]
Comment (abhinandan13jan) [reply src/hooks/usePipelineRuns.ts]: updated

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**